src/main/java/DukeException.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * DukeException class handles the exception specific to Duke chatbot.
+ */
+public class DukeException extends Exception {
+    public String tmp = "";
+    public int num = 0;
+    public String tmp2 = "";
+    public String tmp3 = "";
+
+    public DukeException(String tmp) {
+        super(tmp);
+        this.tmp = tmp;
+    }
+
+    public DukeException(String tmp, int num) {
+        super(tmp);
+        this.tmp = tmp;
+        this.num = num;
+    }
+
+    public DukeException(String tmp, String tmp2, String tmp3) {
+        super(tmp);
+        this.tmp = tmp;
+        this.tmp2 = tmp2;
+        this.tmp3 = tmp3;
+    }
+
+    /**
+     * Prints out the exception.
+     * @return String of the exception.
+     */
+    @Override
+    public String toString() {
```

Comment:
It might be clearer if you implement DukeException as an abstract class and have more specific exception classes extending from DukeException. That way your exceptions can be better distinguished!